test/loco/integer/sendmoremoney2.clj

```diff
@@ -0,0 +1,69 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Send + More = Money example solution #2 + tests. This solution takes the long road of
+            having a var for each letter, constraints on the addends of the sum plus carries and
+            does not make use of loco's $scalar constraint (see sendmoremoney.clj for a more concise
+            implementation)." }
+  loco.integer.sendmoremoney2
+  (:refer-clojure :exclude [send]) 
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]))
+
+(defn initialize-digits [vars]
+  (for [v vars]
+    ($in v 0 9)))
+
+(def letters
+  [:S :E :N :D :M :O :R :Y])
+
+(def carries
+  (for [i (range 4)]
+    ($in [:_c i] 0 1)))
+
+(def ^{:doc "Model #2. Constraints on the addends with carries:
+               c3 c2 c1 c0     
+                S  E  N  D +
+                M  O  R  E =
+             ---------------
+             M  O  N  E  Y  "}
+  smm2-model
+  (concat letters
+          (initialize-digits letters)
+          carries
+          [($distinct letters)
+           ($!= :S 0)
+           ($!= :M 0)
+           ($= ($+         :D :E) ($+ ($* 10 [:_c 0]) :Y))
+           ($= ($+ [:_c 0] :N :R) ($+ ($* 10 [:_c 1]) :E))
+           ($= ($+ [:_c 1] :E :O) ($+ ($* 10 [:_c 2]) :N))
+           ($= ($+ [:_c 2] :S :M) ($+ ($* 10 [:_c 3]) :O))
+           ($=     [:_c 3]                            :M)]))
+
+(defn reorder-solution
+  "Manages to rearrange a solution in a better way."
+  [solution]
+  (array-map :send [(find solution :S) (find solution :E) (find solution :N) (find solution :D)]
+             :more [(find solution :M) (find solution :O) (find solution :R) (find solution :E)]
+             :money [(find solution :M) (find solution :O) (find solution :N) (find solution :E) (find solution :Y)]))
+
+(deftest sendmoremoney-tests
+  "Testing Send + More = Money solution #2"
+
+  (doseq [sol (solutions smm2-model)]
```

Comment:
This seems inconsistent from the other file. In sendmoremoney1 you are only checking the first solution, but in sendmoremoney2 you are checking all of the solutions against one pre-defined solution. I'd like both of them to test that there is ONLY one solution, which matches the desired solution.

In both files this test case could be shortened to just `(is (= (solutions smm2-model) '({:S 9 :E 5 ...})))`.